src/util/orm.py

```diff
@@ -281,35 +281,47 @@ def recompute_fields(cr, model, fields, ids=None, logger=_logger, chunk_size=256
     assert strategy in {"flush", "commit", "auto"}
     Model = env(cr)[model] if isinstance(model, basestring) else model
     model = Model._name
-    if ids is None:
-        cr.execute('SELECT id FROM "%s"' % table_of_model(cr, model))
-        ids = tuple(map(itemgetter(0), cr.fetchall()))
-
-    if strategy == "auto":
-        big_table = len(ids) > BIG_TABLE_THRESHOLD
-        any_tracked_field = any(getattr(Model._fields[f], _TRACKING_ATTR, False) for f in fields)
-        strategy = "commit" if big_table and any_tracked_field else "flush"
-
-    size = (len(ids) + chunk_size - 1) / chunk_size
-    qual = "%s %d-bucket" % (model, chunk_size) if chunk_size != 1 else model
-    for subids in log_progress(chunks(ids, chunk_size, list), logger, qualifier=qual, size=size):
-        records = Model.browse(subids)
-        for field_name in fields:
-            field = records._fields[field_name]
-            if hasattr(records, "_recompute_todo"):
-                # < 13.0
-                records._recompute_todo(field)
-            else:
-                Model.env.add_to_compute(field, records)
 
-        recompute(records)
-        # trigger dependent fields recomputation
-        records.modified(fields)
-        if strategy == "commit":
-            cr.commit()
-        else:
-            flush(records)
-        invalidate(records)
+    def get_record_ids():
+        if ids:
+            yield ids
+            return
+        MAX_SIZE = 1000000
```

Comment:
```suggestion
        MAX_SIZE = 2**20
```
For alignment with the default `chunk_size`?